synapse/push/clientformat.py

```diff
@@ -63,15 +56,12 @@ def format_push_rules_for_user(
         for c in template_rule["conditions"]:
             c.pop("_cache_key", None)
 
-            pattern_type = c.pop("pattern_type", None)
-            if pattern_type == "user_id":
-                c["pattern"] = user.to_string()
-            elif pattern_type == "user_localpart":
-                c["pattern"] = user.localpart
-
-            sender_type = c.pop("sender_type", None)
-            if sender_type == "user_id":
-                c["sender"] = user.to_string()
+            for type_key in ("pattern", "sender", "value"):
```

Comment:
It seems like `pattern` and `sender` is currently unused in the rules predefined by synapse, and user-defined rules can't be generic ones with `_type`.
We should then probably just keep `value_type` here ?

Reply:
`pattern_type` is actually used. A lot ^^
I can't spot a `sender_type` however.